core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -69,8 +70,16 @@ object RequestChannel extends Logging {
       } else
         null
     val body: AbstractRequest =
-      if (requestObj == null)
-        AbstractRequest.getRequest(header.apiKey, header.apiVersion, buffer)
+      if (requestObj == null) {
+        try {
+          // If the request version is higher than supported version, UnsupportedVersionException might be thrown,
```

Comment:
If we add check on min version as well based on #986, we should reflect that in the comment here as well.